gc/default/default.c

```diff
@@ -2834,6 +2834,8 @@ rb_gc_impl_define_finalizer(void *objspace_ptr, VALUE obj, VALUE block)
 
     RBASIC(obj)->flags |= FL_FINALIZE;
 
+    int lev = rb_gc_vm_lock();
+
```

Comment:
I can't add a line comment since you didn't modify that block, but do you have to unlock at 2849 before the early return?

Reply:
Good catch 6d0dd7d86390458e02830684dd45830b5869a7aa